debezium-core/src/main/java/io/debezium/relational/TableSchema.java

```diff
@@ -108,6 +111,16 @@ public Schema keySchema() {
         return keySchema;
     }
 
+    /**
+     * Get the {@link Schema} that represents the table's primary key or the artificial table's primary key for tables not providing primary key.
+     *
+     * @return the Schema describing the all column's that make up the primary key if table has no explicit primary key
+     */
+    @Override
+    public Schema keySchemaOrProxyKeySchema() {
```

Comment:
Is there a strong reason for exposing this information separately, instead of just via the existing `keySchema()` method? After all, that's what's done for MySQL, too. I.e. why is important for calling code to know about the difference? Or can we get rid of this and just feed the surrogate key definition into the schema when building it?

Reply:
@gunnarmorling This was intentional as changing the behaviour of the main method caused issues with existing code. So I opted for incremental approach - provide a new contract for connectors using new framework and keep the behaviour for the existing ones.
When we migrate everything to the new framework the old `keySchema` at all could be removed and the contract simplified.

Reply:
What were these issues? I don't see how this would impact existing code in other connectors: only for SQL Server, the code calling the table schema builder would pass the unique key instead of the PK in this particular case. This should be opaque to any other usages?

Reply:
@gunnarmorling SO, the issues on MySQL side identified and fixed. I've simplified the code as well.